scalr/data_source_scalr_module_version.go

```diff
@@ -46,7 +46,19 @@ func dataSourceModuleVersionRead(ctx context.Context, d *schema.ResourceData, me
 	var version string
 	if v, ok := d.GetOk("version"); ok {
 		version = v.(string)
-		mv, err = scalrClient.ModuleVersions.ReadBySemanticVersion(ctx, module.ID, version)
```

Comment:
Please, remove `ReadBySemanticVersion` from `go-scalr` too.

Reply:
`ReadBySemanticVersion` have `version` validation. Please, add `ValidateFunc` to `version` Schema.

Reply:
made validString and validStringID public, added validation, and removed `ReadBySemanticVersion`

Reply:
to re-use validation and avoid the duplication of the provided line. I believe there will be another cases where it can be used.

Reply:
> I don't see the need to export these functions too. They're primitive and not really very useful, while adding more dependency on client package. We should aim to minimise that, as we're planning to switch to autogenerated client code eventually and `go-scalr` will be dropped entirely.

thank you, took into account all your feedback. Reverted go-scalr changes, left only removing un-used `ReadBySemanticVersion`, used `StringIsNotWhiteSpace` validator